SYCL/Basic/half_builtins.cpp

```diff
@@ -0,0 +1,227 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %HOST_RUN_PLACEHOLDER %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+// OpenCL CPU driver does not support cl_khr_fp16 extension
+// UNSUPPORTED: cpu && opencl
+
+#include <CL/sycl.hpp>
+
+#include <cmath>
+#include <unordered_set>
+
+using namespace cl::sycl;
+
+constexpr int N = 16 * 3; // divisible by all vector sizes

```

Comment:
```suggestion
constexpr int N = 16; // divisible by all vector sizes
```
Looking that macro processing 3-element vector I see that N is not required to be divisible by 3.